lib/Controller/SettingsController.php

```diff
@@ -88,10 +97,18 @@ public function __construct($AppName, IRequest $request,
 	 */
 	public function addClient() {
 		if (!isset($_POST['redirect_uri']) || !isset($_POST['name'])) {
-			return new RedirectResponse('../../settings/admin?sectionid=additional#' . $this->appName);
+			return new RedirectResponse(
+				$this->urlGenerator->linkToRouteAbsolute(
+					'settings.SettingsPage.getAdmin',
+					['sectionid' => 'authentication']
+				) . '#oauth2');
```

Comment:
True, but since the html anchor was hard coded anyway I didn't think it mattered too much :P